<launch>
    <arg name="manager_name"                default="ssd_nodelet_manager" />
    <arg name="manager_threads"             default="4" />

    <arg name="ssd_img_show_flag"           default="true" />
    <arg name="ssd_execute_default"         default="true" />
    <!-- <arg name="ssd_pub_result_image"        default="true" /> -->
    <arg name="rqt_image_view_flag"         default="false" />

    <arg name="ssd_image_topic_name"        default="/camera/rgb/image_raw" />

    <arg name="ssd_in_scale_factor"         default="0.007843" />
    <arg name="ssd_confidence_threshold"    default="0.5" />

    <arg name="ssd_prototxt_name"           default="voc_object.prototxt" />
    <arg name="ssd_caffemodel_name"         default="voc_object.caffemodel" />
    <arg name="ssd_class_names_file"        default="voc_object_names.txt" />

    <arg name="object_specified_enabled"     default="true" />
    <arg name="specified_object_name"        default="person" />

    <group ns = "ssd_object_detect">
        <node pkg="nodelet" type="nodelet" name="$(arg manager_name)" args="manager" output="screen">
            <param name="num_worker_threads" value="$(arg manager_threads)" />
        </node>

        <node pkg="nodelet" type="nodelet" name="ssd_nodelet" args="load ssd_nodelet/ImageSubscriber $(arg manager_name)" >
            <param name="ssd_img_show_flag"         type="bool"     value="$(arg ssd_img_show_flag)"/>
            <param name="ssd_execute_default"       type="bool"     value="$(arg ssd_execute_default)"/>
            <!-- <param name="ssd_pub_result_image"      type="bool"     value="$(arg ssd_pub_result_image)"/> -->

            <param name="ssd_image_topic_name"      type="str"      value="$(arg ssd_image_topic_name)"/>

            <param name="ssd_in_scale_factor"       type="double"   value="$(arg ssd_in_scale_factor)"/>
            <param name="ssd_confidence_threshold"  type="double"   value="$(arg ssd_confidence_threshold)"/>

            <param name="ssd_prototxt_name"         type="str"      value="$(arg ssd_prototxt_name)"/>
            <param name="ssd_caffemodel_name"       type="str"      value="$(arg ssd_caffemodel_name)"/>
            <param name="ssd_class_names_file"      type="str"      value="$(arg ssd_class_names_file)"/>

            <param name="object_specified_enabled"   type="bool"     value="$(arg object_specified_enabled)"/>
            <param name="specified_object_name"      type="str"      value="$(arg specified_object_name)"/>

        </node>
    </group>
    <node if="$(arg rqt_image_view_flag)" pkg="rqt_image_view" type="rqt_image_view" name="image_view_debug" args="/ssd_object_detect/detect_result"/>
</launch>
